server/types/app.go

```diff
@@ -72,6 +74,10 @@ type (
 	// application using various configurations.
 	AppCreator func(log.Logger, dbm.DB, io.Writer, AppOptions) Application
 
+	// TestnetAppCreator serves a similar purpose as AppCreator, however is used for
+	// passing more information to the application layer needed to create a testnet from existing state.
+	TestnetAppCreator func(log.Logger, dbm.DB, io.Writer, bytes.HexBytes, crypto.PubKey, string, AppOptions) Application
```

Comment:
Could we pass more information as `AppOptions` to avoid duplicating API and, as a result, parameters to `startInProcess`?

Reply:
Really good suggestion, am implementing now. 

I thought it wasn't possible to type case the viper inputs that come from AppOptions to anything other than native types but that is not the case.